playground/src/Playground.tsx

```diff
@@ -40,30 +53,55 @@ type PlaygroundProps = {
 /**
   Used to render your component inside a live-editable playground and directly see the output of the code used.
 */
-export const Playground = ({ code, scope, noInline }: PlaygroundProps) => {
+export const Playground = ({
+  useShadowDOM,
+  styles,
+  code,
+  scope,
+  noInline,
+}: PlaygroundProps) => {
   const commentsRegex = /([^http:|https:]\/\/.*)|(\/\*(.|\n)*?\*\/)/gm;
-  return (
+  const template = (
     <LiveProvider
       code={code.trim()}
       scope={scope}
       noInline={noInline}
       theme={oceanicNext}
       transformCode={(code) => {
         const codeWithoutComments = code.replaceAll(commentsRegex, '').trim();
-        return codeWithoutComments.startsWith('<') ? `<>${codeWithoutComments}</>` : codeWithoutComments;
+        return codeWithoutComments.startsWith('<')
+          ? `<>${codeWithoutComments}</>`
+          : codeWithoutComments;
       }}
     >
       <div>
-        <div style={styles.preview} id="playground-preview">
+        <div style={playgroundStyles.preview}>
           <LivePreview />
         </div>
-        <LiveEditor style={styles.editor} />
-        <LiveError style={styles.error} />
+        <LiveEditor style={playgroundStyles.editor} />
+        <LiveError style={playgroundStyles.error} />
       </div>
     </LiveProvider>
   );
+
+  if (!useShadowDOM) {
+    return template;
+  }
+
+  const playgroundRef = useRef();
+
+  useEffect(() => {
+    const playgroundElement = playgroundRef.current as DockitReactPlayground;
+    if (!playgroundElement) return;
+    playgroundElement.styles = styles;
+    ReactDOM.render(template, playgroundElement.$renderRoot);
+  });
+
+  return (
+    <dockit-react-playground ref={playgroundRef}></dockit-react-playground>
```

Comment:
I get annoyed but TS error

> Property 'dockit-react-playground' does not exist on type 'JSX.IntrinsicElements'.(2339)

any tips on how to properly fix it in the context of `dockit-react`?

Reply:
doesn't happen locally in VSCode
smth specific to our Monaco setup of TS?